server.go

```diff
@@ -187,48 +181,55 @@ func (u *Upgrader) Upgrade(w http.ResponseWriter, r *http.Request, responseHeade
 		c.newDecompressionReader = decompressNoContextTakeover
 	}
 
-	p := c.writeBuf[:0]
-	p = append(p, "HTTP/1.1 101 Switching Protocols\r\nUpgrade: websocket\r\nConnection: Upgrade\r\nSec-WebSocket-Accept: "...)
-	p = append(p, computeAcceptKey(challengeKey)...)
-	p = append(p, "\r\n"...)
+	// workaround for haxe in newConnBRW
+	brw.Writer.Reset(netConn)
```

Comment:
We should dig into why this is necessary as a separate issue. When they call and use writeHook, it writes out a single byte (zero) and Flushes into the buffer. I don't immediately see why this is done.